deepspeed/launcher/launch.py

```diff
@@ -227,8 +259,28 @@ def main():
 
             # spawn the processes
             cmd = []
+            if args.bind_cores_to_rank:
+                if args.bind_core_list != None:
+                    core_list = parse_range_list(args.bind_core_list)
+                    total_cores = len(core_list)
+                else:
+                    total_cores = psutil.cpu_count(logical=False)
+                    core_list = range(total_cores)
+                cores_per_rank = total_cores // num_local_procs
+                assert cores_per_rank >= 1, "At least one core needs to be assigned to each rank"
+                core_list_for_rank = core_list[cores_per_rank *
+                                               local_rank:cores_per_rank *
+                                               (local_rank + 1)]
+                os.environ["OMP_NUM_THREADS"] = f"{cores_per_rank}"
+                cmd.append("numactl")
```

Comment:
Great question!

Reply:
@delock has added a `numactl` availability check and error message, which is probably fine for now.

Reply:
Availablility check had been made according to libaio example.  The new check for numactl will be a 'soft' one since the check is only for popular distros.